packages/text-input/src/TextInput.tsx

```diff
@@ -106,10 +106,14 @@ interface BaseTextInputProps extends HTMLElementProps<'input'> {
   darkMode?: boolean;
 
   type?: TextInputType;
+
+  ['aria-labelledby']?: string;
 }
 
+type TextInputProps = BaseTextInputProps &
+  Omit<HTMLElementProps<'input', never>, keyof BaseTextInputProps>;
 type AriaLabels = 'label' | 'aria-labelledby';
-type TextInputProps = Either<BaseTextInputProps, AriaLabels>;
+type AccessibleTextInputProps = Either<TextInputProps, AriaLabels>;
```

Comment:
For some reason when I type an object as:

`const test: AccessibleTextInputProps = {}`

I see an error that there's no label; however, no error is raised when you write `<TextInput />` without passing a label prop. Was wondering if I could get your 👀 @lazytype

Reply:
I'm willing to bet that `React.forwardRef` is causing the props to lose some type information

Reply:
Think it's resolved

Reply:
Nvm!